helm-charts/mach-operator/templates/statefulset.yaml

```diff
@@ -0,0 +1,123 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "mach-operator.fullname" . }}
+  labels:
+    {{- include "mach-operator.labels" . | nindent 4 }}
+    {{- with .Values.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end}}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "mach-operator.selectorLabels" . | nindent 6 }}
+        {{- with .Values.labels }}
+        {{- toYaml . | nindent 6 }}
+        {{- end}}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        app: {{ include "mach-operator.fullname" . }}
+        {{- include "mach-operator.selectorLabels" . | nindent 8 }}
+        {{- with .Values.labels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end}}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "mach-operator.fullname" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: node
+          {{- with .Values.node.args }}
+          args:
+          {{- toYaml . | nindent 12 }}
+          {{- end }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.node.image.repository }}:{{ .Values.node.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.node.image.pullPolicy }}
+          ports:
+          {{- toYaml $.Values.ports | nindent 10 }}
+          {{- if .Values.lifecycleHooks }}
+          lifecycle:
+          {{- toYaml .Values.node.lifecycleHooks | nindent 12 }}
+          {{- end }}
+          {{- if .Values.node.livenessProbe }}
+          livenessProbe:
+            {{- toYaml .Values.node.livenessProbe | nindent 12 }}
+          {{- end }}
+          {{- if .Values.node.readinessProbe }}
+          readinessProbe:
+            {{- toYaml .Values.node.readinessProbe | nindent 12 }}
+          {{- end }}
+          resources:
+            {{- toYaml .Values.node.resources | nindent 12 }}
+          volumeMounts:
+            - name: config
```

Comment:
Can you make it optional via default `values.yaml` maybe? Because in my case I'd prefer to import bls key from a separate secret not related to chart, for example.

Reply:
How about this?

```
{{- define "mach-operator.secretsName" -}}
    {{- if .Values.secrets.create -}}
        {{ default (include "mach-operator.fullname" .) .Values.secrets.name }}
    {{- else -}}
        {{ default "default" .Values.secrets.name }}
    {{- end -}}
{{- end -}}
```

Reply:
It could also be provided with Vault Injector directly to the file, but you have the mount for the current one. But for me it's ok :)

Reply:
ok